src/plugins/network/network.c

```diff
@@ -50,6 +52,93 @@ int elektraNetworkAddrInfo (Key * toCheck)
 	return 0;
 }
 
+int elektraPortInfo (Key * toCheck, Key * parentKey)
+{
+	const Key * meta = keyGetMeta (toCheck, "check/port");
+	const Key * listenMeta = keyGetMeta (toCheck, "check/port/listen");
+	if (!meta && !listenMeta) return 0; /* No check to do */
+	char * endptr = NULL;
+	long portNumber = strtol (keyString (toCheck), &endptr, 10);
+	int portNumberNetworkByteOrder;
+
+	if (*endptr == '\0')
+	{
+		if (portNumber < 0 || portNumber > 65535)
+		{
+			ELEKTRA_SET_ERRORF (171, parentKey, "Port %ld on key %s was not within 0 - 65535", portNumber, keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = htons (portNumber);
+	}
+	else
+	{
+		struct servent * service;
+		service = getservbyname (keyString (toCheck), NULL); // NULL means we accept both tcp and udp
+		if (service == NULL)
+		{
+			ELEKTRA_SET_ERRORF (205, parentKey, "Could not find service with name %s on key %s", keyString (toCheck),
+					    keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = service->s_port;
+	}
+
+	if (!listenMeta) return 0; /* No check to do */
+
+	char * hostname = "localhost";
+
+	int sockfd;
+	struct sockaddr_in serv_addr;
+	struct hostent * server;
+	sockfd = socket (AF_INET, SOCK_STREAM, 0);
+	if (sockfd < 0)
+	{
+		ELEKTRA_SET_ERRORF (205, parentKey, "Could not open a socket: %s", strerror (errno));
+	}
+
+	server = gethostbyname (hostname);
+	if (server == NULL)
+	{
+		if (errno == HOST_NOT_FOUND)
+		{
+			ELEKTRA_SET_ERRORF (205, parentKey, "Could not connect to %s: No such host", hostname);
+			return -1;
+		}
+		else
+		{
+			ELEKTRA_SET_ERRORF (205, parentKey, "There was an error when trying to connect to host %s . errno: %s", hostname,
```

Comment:
Since the error message already uses the human readable form of `errno`, I think it would make sense to not mention the term `errno` at all. How about

```c
"There was an error trying to connect to host %s: %s"
```

instead of

```c
"There was an error when trying to connect to host %s . errno: %s"
```

?